postman-generator/src/main/scala/generator/PostmanCollectionGenerator.scala

```diff
@@ -92,38 +93,38 @@ object PostmanCollectionGenerator extends CodeGenerator {
       )
     }
 
-    val cleanupFolder: postman.Folder = PredefinedCollectionItems.prepareCleanupFolder()
-    val setupFolder: postman.Folder = PredefinedCollectionItems.prepareSetupFolder()
-
-    val objReferenceAttrToOperationTuples = DependantOperationResolver.resolve(resolvedService)
-    val requiredEntitiesSetupSteps = objReferenceAttrToOperationTuples.map {
-      case (objRefAttr, operation) =>
-        val postmanItem = PostmanItemBuilder.build(operation, serviceSpecificHeaders, examplesProvider, None)
-        val postmanItemWithTests = addItemTests(postmanItem, None)
-
-        addDependencyItemVarSetting(objRefAttr, postmanItemWithTests)
-    }
-
-    val setupFolderWithDependantEntities = setupFolder.copy(
-      item = setupFolder.item ++ requiredEntitiesSetupSteps
-    )
-
     val basicAuthOpt = service.attributes
-      .find(_.name.equalsIgnoreCase(PostmanBasicAuthAttribute.Key))
+      .find(_.name.equalsIgnoreCase(PostmanAttributes.BasicAuthKey))
       .flatMap(_.value.asOpt[PostmanBasicAuthAttrValue])
       .map { basicAuth =>
-      postman.Auth(
-        `type` = postman.AuthEnum.Basic,
-        basic = Some(List(
-          postman.BasicAuth(key = "username", value = basicAuth.username),
-          postman.BasicAuth(key = "password", value = basicAuth.password)
-        ))
-      )
-    }
+        postman.Auth(
+          `type` = postman.AuthEnum.Basic,
+          basic = Some(List(
+            postman.BasicAuth(key = "username", value = basicAuth.username),
+            postman.BasicAuth(key = "password", value = basicAuth.password)
+          ))
+        )
+      }
+
+    val shouldAddSetupFolder: Boolean =
+      service.attributes
+        .find(_.name.equalsIgnoreCase(PostmanAttributes.SetupKey))
+        .nonEmpty
+
+    val entitiesSetupFolderOpt = prepareEntitiesSetupFolders(resolvedService, serviceSpecificHeaders, examplesProvider)
+
+    val folders: Seq[Folder] =
+      (if (shouldAddSetupFolder) {
+        val cleanupFolder: postman.Folder = PredefinedCollectionItems.prepareCleanupFolder()
+        val setupFolder: postman.Folder = PredefinedCollectionItems.prepareSetupFolder()
+        (Option(setupFolder) :: entitiesSetupFolderOpt :: Nil).flatten ++ postmanCollectionFolders.toList ++ List(cleanupFolder)
```

Comment:
`Option(setupFolder)` - option seems redundant

Reply:
okay, I see what you did here - maybe `Some` will be more readable?

Reply:
using some is unsafe, it could be null 😄